### ISSUE TITLE
ValueError: math domain error

### ISSUE THREAD

**Issue by hoetzgit** [chaunceygardiner/weewx-loopdata#10]
Hello, my English is poor and automatically translated, but I'll paste the following error message here. If you need details, feel free to contact me. I will try to give them to you then.

Many greetings
Henry

```
Dec  8 09:01:46 weewx weewx[1502695] CRITICAL user.loopdata:     ****  Traceback (most recent call last):
Dec  8 09:01:46 weewx weewx[1502695] CRITICAL user.loopdata:     ****    File "/home/weewx/bin/user/loopdata.py", line 1577, in process_queue
Dec  8 09:01:47 weewx weewx[1502695] CRITICAL user.loopdata:     ****      loopdata_pkt = LoopProcessor.generate_loopdata_dictionary(pkt, self.cfg, self.accumulators)
Dec  8 09:01:47 weewx weewx[1502695] CRITICAL user.loopdata:     ****    File "/home/weewx/bin/user/loopdata.py", line 1682, in generate_loopdata_dictionary
Dec  8 09:01:47 weewx weewx[1502695] CRITICAL user.loopdata:     ****      return LoopProcessor.create_loopdata_packet(pkt, cfg, accums)
Dec  8 09:01:47 weewx weewx[1502695] CRITICAL user.loopdata:     ****    File "/home/weewx/bin/user/loopdata.py", line 1914, in create_loopdata_packet
Dec  8 09:01:48 weewx weewx[1502695] CRITICAL user.loopdata:     ****      LoopProcessor.add_period_obstype(cname,  accum, loopdata_pkt, cfg.converter, cfg.formatter)
Dec  8 09:01:48 weewx weewx[1502695] CRITICAL user.loopdata:     ****    File "/home/weewx/bin/user/loopdata.py", line 1776, in add_period_obstype
Dec  8 09:01:48 weewx weewx[1502695] CRITICAL user.loopdata:     ****      src_value = stats.rms
Dec  8 09:01:48 weewx weewx[1502695] CRITICAL user.loopdata:     ****    File "/home/weewx/bin/user/loopdata.py", line 468, in rms
Dec  8 09:01:48 weewx weewx[1502695] CRITICAL user.loopdata:     ****      return math.sqrt(self.wsquaresum / self.sumtime) if self.count else None
Dec  8 09:01:48 weewx weewx[1502695] CRITICAL user.loopdata:     ****  ValueError: math domain error

```

**Comment by chaunceygardiner** [comment]
Hi Henry,I think I see the issue.  I uploaded a fix to github but did not do a release.  Can you try the new bin/user/loopdata.py file or do I need to do a release?BTW, what driver are you using?Cheers,JohnOn Dec 8, 2022, at 2:43 AM, hoetzgit ***@***.***> wrote:﻿
Hello, my English is poor and automatically translated, but I'll paste the following error message here. If you need details, feel free to contact me. I will try to give them to you then.
Many greetings
Henry
Dec  8 09:01:46 weewx weewx[1502695] CRITICAL user.loopdata:     ****  Traceback (most recent call last):
Dec  8 09:01:46 weewx weewx[1502695] CRITICAL user.loopdata:     ****    File "/home/weewx/bin/user/loopdata.py", line 1577, in process_queue
Dec  8 09:01:47 weewx weewx[1502695] CRITICAL user.loopdata:     ****      loopdata_pkt = LoopProcessor.generate_loopdata_dictionary(pkt, self.cfg, self.accumulators)
Dec  8 09:01:47 weewx weewx[1502695] CRITICAL user.loopdata:     ****    File "/home/weewx/bin/user/loopdata.py", line 1682, in generate_loopdata_dictionary
Dec  8 09:01:47 weewx weewx[1502695] CRITICAL user.loopdata:     ****      return LoopProcessor.create_loopdata_packet(pkt, cfg, accums)
Dec  8 09:01:47 weewx weewx[1502695] CRITICAL user.loopdata:     ****    File "/home/weewx/bin/user/loopdata.py", line 1914, in create_loopdata_packet
Dec  8 09:01:48 weewx weewx[1502695] CRITICAL user.loopdata:     ****      LoopProcessor.add_period_obstype(cname,  accum, loopdata_pkt, cfg.converter, cfg.formatter)
Dec  8 09:01:48 weewx weewx[1502695] CRITICAL user.loopdata:     ****    File "/home/weewx/bin/user/loopdata.py", line 1776, in add_period_obstype
Dec  8 09:01:48 weewx weewx[1502695] CRITICAL user.loopdata:     ****      src_value = stats.rms
Dec  8 09:01:48 weewx weewx[1502695] CRITICAL user.loopdata:     ****    File "/home/weewx/bin/user/loopdata.py", line 468, in rms
Dec  8 09:01:48 weewx weewx[1502695] CRITICAL user.loopdata:     ****      return math.sqrt(self.wsquaresum / self.sumtime) if self.count else None
Dec  8 09:01:48 weewx weewx[1502695] CRITICAL user.loopdata:     ****  ValueError: math domain error

**Comment by hoetzgit** [comment]
Hi, is changed and restarted WeeWX. Thanks for the change, I will watch it. 

I try a lot with WeeWX and extensions, but have quite little idea about Python and weather in general. Poor is unfortunately also my English and I have to translate everything to me. But it is fun. 

If you want to see my configuration (driver etc), I have a [GitHub repository](https://github.com/hoetzgit/weewx-pwsweiherhammer).

I will let you know if the bug is fixed now.

Many greetings
Henry

**Comment by chaunceygardiner** [comment]
Hi Henry,

I made a change to the file I sent you and have released a fix as version v3.3.1.  Thanks for reporting this.
